lib/messages.js

```diff
@@ -3,6 +3,34 @@ var _ = require('underscore'),
     utils = require('./utils'),
     logger = require('./logger');
 
+function extendedTemplateContext(doc, extras) {
```

Comment:
nit : can you either document what fields are expected in `extras`, or pass them as individual arguments? I'm fighting against the extras/options pattern, it makes things hard to follow. (types!)